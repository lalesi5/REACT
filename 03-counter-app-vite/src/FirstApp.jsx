import React from 'react';
import PropTypes from 'prop-types';

// const getValue = () => {
//     return 4+4;
// }

export const FirstApp = ({title, subtitle, name}) => {

// console.log(props);

    return (
        <>
            <h1 data-testid="test-id">{title}</h1>
            {/* <code>{getValue()}</code> */}
            <h2>{subtitle}</h2>
            <a>{name}</a>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Alex Carrera',
    subtitle: 'No hay subtitulo',
    title: 'No hay titulo',
}