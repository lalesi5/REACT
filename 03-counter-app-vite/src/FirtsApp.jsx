import PropTypes from 'prop-types';

// const getValue = () => {
//     return 4+4;
// }

export const FirtsApp = ({title, subtitle, name}) => {

// console.log(props);

    return (
        <>
            <h1>{title}</h1>
            {/* <code>{getValue()}</code> */}
            <p>{subtitle}</p>
            <a>{name}</a>
        </>
    )
}


FirtsApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}

FirtsApp.defaultProps = {
    name: 'Alex Carrera',
    subtitle: 124,
    title: 'No hay titulo',
}