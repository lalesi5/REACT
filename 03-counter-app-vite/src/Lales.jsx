export const Lales = () => {
  return (
    <div>Esta es una prueba LaLeS</div>
  )
}
