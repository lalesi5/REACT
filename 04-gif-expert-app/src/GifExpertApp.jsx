import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = (newCaterory) => {

        if (categories.includes(newCaterory)) return;
        //console.log(newCaterory);
        setCategories([newCaterory,...categories ]);
        // setCategories(cat => [...cat, nuevoValor]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onNewValue={(value) => onAddCategory(value)}
            />

            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }

        </>
    )
}
