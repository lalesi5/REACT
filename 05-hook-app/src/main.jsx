//import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHool'
import { MultipleCustomHook } from './03-examples/MultipleCustomHook'

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  <MultipleCustomHook />
  // </React.StrictMode>,
)
