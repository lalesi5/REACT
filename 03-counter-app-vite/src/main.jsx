import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp'

import './styles.css'
import { CounterApp } from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Hola lales'/> */}
        <CounterApp value={100} />
    </React.StrictMode>
)

