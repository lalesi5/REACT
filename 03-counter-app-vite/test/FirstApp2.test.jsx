import { render, screen } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en FirstApp', () => {

    const title = 'Hola lales';
    const subtitle = 'Hola subtitulo';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();

    });

    test('debe mostrar el mensaje "Hola lales"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe demostrar el subtitulo enviado por props', () => {

        render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );

        expect( screen.getAllByText(subtitle).length).toBe(1);

    })

});