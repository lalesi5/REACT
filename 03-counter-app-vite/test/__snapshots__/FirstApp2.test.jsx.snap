// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-id"
  >
    Hola lales
  </h1>
  <h2>
    No hay subtitulo
  </h2>
  <a>
    Alex Carrera
  </a>
</div>
`;
